### PR TITLE
[mono] Add old design docs from mono website.

### PR DESCRIPTION
**Pull request by vargaz**
The original location of these documents is:

https://github.com/mono/website/tree/gh-pages/docs/advanced/runtime/docs

Changes made:
- renamed BITCODE.md -> bitcode.md
- removed some documents like xdebug.md which are no longer relevant

### CODE REVIEW

**Review by lambdageek — APPROVED** [review]

**Comment by lambdageek** [comment]
@vargaz maybe put them in a subdirectory with a README.md that they came from the mono project website and are primarily describing the .NET Framework version of mono?

File: docs/design/mono/web/ascii-strings.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+title: ASCII Mono
+---
+
+## Introduction
+
+This is a proposal for an optimisation to `System.String`.
```
Comment (jkotas) [inline docs/design/mono/web/ascii-strings.md:7]: This describes abandoned proposal that was never productized. I would omit this doc.

We have https://github.com/dotnet/runtime/issues/6612 about this idea that discusses all sorts of problems that would have to be solved to make this idea productized.

File: docs/design/mono/web/coop-suspend.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
+---
+title: Runtime Cooperative Suspend
+---
```
Comment (lambdageek) [inline docs/design/mono/web/coop-suspend.md:3]: Some of this is also covered in 
 https://github.com/dotnet/runtime/blob/main/docs/design/mono/mono-thread-state-machine.md

but some of the info is only here (for example: `MONO_REQ_GC_UNSAFE` etc)

File: docs/design/mono/web/coop-suspend.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+* AsyncSuspended: Thread was async suspended, so it's on a signal handler or thread_suspend was called on it. (This state never happens when running threads are cooperatively suspended)
+* Blocking: The current thread is executing code that won't touch managed memory.  There are no pending suspend requests.
+* BlockingSuspendRequested: The current thread is executing code that won't touch managed memory, and someone requested it to suspend.  In full cooperative mode, the thread is assumed to still be suspended.
+* BlockingSelfSuspended: The current thread finished executing blocking code but there was a pending suspend against it, it's waiting to be resumed.
+* BlockingAsyncSuspended: The current thread was executing in blocking code, but it was preemptively suspended.  This is done in "hybrid" suspend mode.  When the thread resumes, it will go back to executing blocking code.
+
+![Coop state machine transition diagram](/docs/advanced/runtime/docs/coop-state-machine.png)
+
```
Comment (lambdageek) [inline docs/design/mono/web/coop-suspend.md:76]: Do the other docs also have missing pictures?

File: docs/design/mono/web/other.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+$ csc fib.cs
+$ MONO_JIT_DUMP_METHOD=Fib::fib mono fib.exe
+cfg_dump: create context for "Fib::fib"
+```
+
+now switch to IGV, you should see something like that: [![igv-screenshot.png](/images/igv-screenshot.png)](/images/igv-screenshot.png)
+
+you can explore the different compiler passes in the navigation bar on the left side. IGV also has a graph diff feature:
+
+[![igv-diff.png](/images/igv-diff.png)](/images/igv-diff.png)
```
Comment (lambdageek) [inline docs/design/mono/web/other.md:87]: missing pics

File: docs/design/mono/web/soft-debugger.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+ The Mono Soft Debugger first became available with Mono 2.6 and is primarily used today with [Mono on the iPhone](http://monotouch.net) and is used from the [MonoDevelop IDE](http://monodevelop.com).
+
+Architecture
+------------
+
+The following diagram is useful in the discussion of the soft debugger:
+
+[![0911030528Mp6F5SHL.png](/archived/images/2/2a/0911030528Mp6F5SHL.png)](/archived/images/2/2a/0911030528Mp6F5SHL.png)
+
```
Comment (lambdageek) [inline docs/design/mono/web/soft-debugger.md:20]: missing pics

**Review by lambdageek — APPROVED** [review]

File: docs/design/mono/web/aot.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
+---
+title: Ahead of Time Compilation (AOT)
+redirect_from:
+  - /Mono%3ARuntime%3ADocumentation%3AAOT/
+---
```
Comment (mdh1418) [inline docs/design/mono/web/aot.md:5]: Do these title/redirect_from still make sense to have? It also doesn't seem to format properly, if they don't add value for the docs here, can they be removed to reduce confusion?

File: docs/design/mono/web/bitcode.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
+---
+title: Bitcode
+---
+
+## Introduction
+
+Bitcode imposes the following major restrictions:
```
[the remark this answers is not in the document]
Comment (vargaz) [reply docs/design/mono/web/bitcode.md:5]: The same backend is used for webassembly.

**Pull request merged**